ocaml-lsp-server/src/custom_requests/req_dune_contexts.ml

```diff
@@ -0,0 +1,49 @@
+open Import
+
+let capability = ("handleDuneContexts", `Bool true)
+
+let meth = "ocamllsp/duneContexts"
+
+let on_request () =
+  match Bin.which "dune" with
+  | None ->
+    Jsonrpc.Response.Error.raise
+      (Jsonrpc.Response.Error.make
+         ~code:InternalError
+         ~message:"dune binary not found"
+         ())
+  | Some dune -> (
+    let describe =
+      "DUNE_CONFIG__GLOBAL_LOCK=disabled " ^ Fpath.to_string dune
```

Comment:
`DUNE_CONFIG__GLOBAL_LOCK=disabled`
^ :(

but the question is — is `dune describe contexts` command's output stable?

Reply:
I see the need for `DUNE_CONFIG__GLOBAL_LOCK` as a bug / limitation of Dune, rather than a hack in this PR. The main issue is that Dune doesn't differentiate between commands that write in `_build` vs commands that only read. So it will always lock the folder regardless. I opened an issue to discuss making the lock mechanism smarter: https://github.com/ocaml/dune/issues/10430

> but the question is — is `dune describe contexts` command's output stable?

This subcommand is being added in the companion PR https://github.com/ocaml/dune/pull/10324. It only outputs the names of the available contexts, so it shouldn't need to change over time.